src/Momento.ts

```diff
@@ -115,9 +118,78 @@ export class Momento {
     });
   }
 
+  public async createSigningKey(
+    ttlMinutes: number,
+    endpoint: string
+  ): Promise<CreateSigningKeyResponse> {
+    this.validateTtlMinutes(ttlMinutes);
+    const request = new control.control_client._CreateSigningKeyRequest();
+    request.ttl_minutes = ttlMinutes;
+    return await new Promise<CreateSigningKeyResponse>((resolve, reject) => {
+      this.client.CreateSigningKey(
+        request,
+        {interceptors: this.interceptors},
+        (err, resp) => {
+          if (err) {
+            reject(cacheServiceErrorMapper(err));
+          } else {
+            resolve(new CreateSigningKeyResponse(endpoint, resp));
+          }
+        }
+      );
+    });
```

Comment:
nit: Instead of copy-pasting this block around you could make a higher order function.

Reply:
That's true, I can do that